cookbook/doctrine/registration_form.rst

```diff
@@ -233,28 +231,43 @@ of the ``User`` class.
 Handling the Form Submission
 ----------------------------
 
-Next, you need a controller to handle the form. Start by creating a simple
-controller for displaying the registration form::
+Next, you need a controller to handle the form rendering and submission. If the
+form is submitted, the controller performs the validation and saves the data
+into the database::
 
-    // src/Acme/AccountBundle/Controller/AccountController.php
-    namespace Acme\AccountBundle\Controller;
+    // src/AppBundle/Controller/AccountController.php
+    namespace AppBundle\Controller;
 
     use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+    use Symfony\Component\HttpFoundation\Request;
+    use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
 
-    use Acme\AccountBundle\Form\Type\RegistrationType;
-    use Acme\AccountBundle\Form\Model\Registration;
+    use AppBundle\Form\Type\RegistrationType;
+    use AppBundle\Form\Model\Registration;
 
     class AccountController extends Controller
     {
-        public function registerAction()
+        /**
+         * @Route("/register", name="account_register")
+         */
+        public function registerAction(Request $request)
         {
-            $registration = new Registration();
-            $form = $this->createForm(new RegistrationType(), $registration, array(
-                'action' => $this->generateUrl('account_create'),
-            ));
+            $form = $this->createForm(new RegistrationType(), new Registration());
+
+            $form->handleRequest($request);
+
+            if ($form->isSubmitted() && $form->isValid()) {
```

Comment:
Maybe we could add a short sentence saying that `->isSubmitted()` is not necessary from a technical point of view (`->isValid()` is enough). But we add it because it makes the code more readable. What do you think?